interview-questions.md

```diff
@@ -7,44 +7,78 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. As a developer, I am creating a Rails application with a model called Cohort that has_many students, but OOPS! I forgot to add the foreign key. How can I fix this mistake? What is the name of the foreign key? Would the foreign key be on the Cohort model or the Student model?
 
-  Your answer:
+  Your answer: cohort model runs through students id and db migrate the new foreign key into the data
 
-  Researched answer:
+  Researched answer: a change can be made by running the rails generate migration. adding a column add column students, cohort id ,integer. then running db migrate
 
 
 
 2. Which RESTful routes must always be passed params? Why?
 
-  Your answer:
+  Your answer: urls? the params must always be passed because its telling the page what information to display
 
-  Researched answer:
+  Researched answer: RESTful routes are always passed parameters because the parameters are linked to the url and helps to get that data and diplay it or perform an action
 
 
 
 3. Name three rails generator commands. What is created by each?
 
-  Your answer:
+  Your answer: rails generate adds data, rails generate controller to make a controller, and mirgrate that interacts with the data 
 
-  Researched answer:
+  Researched answer: 
 
+  rails generate adds data records, adds rows to the table, find data, update data, and remove data.
 
+  rails generate controller the logical center of the application. It connects the interaction between the user, view, and the model. responsible for routing external requests.
 
-4. Consider the Rails routes below. What is the name of the controller method that would be called by each route? What action would each of the controller methods perform?
+  rails generate migration interacts with the data allowing you to evolve your data base. instead of making modifications in SQL.
+
+
+
+
+4. Consider the Rails routes below. What is the name of the controller method that would be called by each route? What action would each of the controller methods perform? 
 
 action: "GET"    location: /students          
 
-action: "POST"   location: /students       
+shows a list of all students entered in the database
+
+action: "POST"   location: /students         
+
+creates a new form of data in the name of students
```

Comment:
Post is the action of actually adding new content to the database.

Reply:
I probably worded this incorrectly, I have a bad problem with it. Just complicating how I explain some things, because its how i've worked it out in my head.